packages/playback-core/src/errors.ts

```diff
@@ -1,9 +1,53 @@
+// Identifies what kind of request was made that resulted in an error
+export const MuxErrorCategory = {
+  VIDEO: 'video',
+  THUMBNAIL: 'thumbnail',
+  STORYBOARD: 'storyboard',
+  DRM: 'drm',
+} as const;
+
+export const MuxErrorCode = {
+  NOT_AN_ERROR: 0,
```

Comment:
NOTE: I implemented unique codes for unique errors for elements, which we can use at various "levels" to translate into additional info (namely, the error dialog for mux player). With the new architecture, we rely on the newly added error category (i.e. was it e.g. DRM or Video, defined above) + error code (i.e. was it a network error or an encrypted error) to provide human readable info (i.e. "Your drm-token was expired")